receiver/gitproviderreceiver/internal/scraper/githubscraper/github_scraper.go

```diff
@@ -108,12 +108,45 @@ func (ghs *githubScraper) scrape(ctx context.Context) (pmetric.Metrics, error) {
 		go func() {
 			defer wg.Done()
 
-			count, err := ghs.getBranches(ctx, genClient, name, trunk)
+			branches, count, err := ghs.getBranches(ctx, genClient, name, trunk)
 			if err != nil {
 				ghs.logger.Sugar().Errorf("error getting branch count for repo %s", zap.Error(err), repo.Name)
 			}
 			ghs.mb.RecordGitRepositoryBranchCountDataPoint(now, int64(count), name)
 
+			// Iterate through the branches populating the Branch focused
+			// metrics
+			for _, branch := range branches {
+				// Check if the branch is the default branch or if it is not behind the default branch
+				if branch.Name == branch.Repository.DefaultBranchRef.Name || branch.Compare.BehindBy == 0 {
```

Comment:
@crobert-1 I'm still running through a few other things, but this comment has been updated to reflect the reasoning here. 

I'm thinking that it might be good for me to add a `limitations` section within the docs for receiver under the GitHub section. These limitations are different for each vendor. For example, when we were developing the GitLab scraper for this receiver, we weren't able to populate the Approval Time metric because it's not exposed via either the REST nor the GraphQL APIs. Thoughts here?

Reply:
Added a limitations section.